### DISCUSSION TITLE
Free Tile Hosting: Public S3 Bucket with CDN

### DISCUSSION THREAD

**Discussion by ganganstyle** [maplibre/maplibre-gl-js#2056] [Ideas]
Perhaps the most performant way to host tiles, is to have an S3 bucket with a CDN.
In the S3 bucket, every tile has its own .pbf file.
This way, every tile request is just a pure data response, with no scripts of any kind.

How about using some of the funds for this project to fund a public bucket, free for everyone to read OSM data from?
There would be no need for credential checks, and every project would contribute to caching.
Imagine the response times!

Maybe it sounds like a crazy idea at first, but it would be a sensational advantage!
S3 buckets and CDNs are known for being extremely cheap, and with e.g. Amazon as a platinum sponsor of the project, maybe this could be the perfect contribution?

**Comment by wipfli** [comment]
Thanks for this suggestion @ganganstyle. I understand that a free vector tiles service would be really convenient, but probably MapLibre is not the right place for this. I think we should focus on open-source software for rendering maps and the donated funds should be use for making the best map renderer in the world.

Maybe some day a company will decided to offer map tiles on their infrastructure for free, who knows, but I don't think that we have the capacity for such a service...

**Reply by wipfli** [reply]
This is what the MapLibre Charter says:
> The MapLibre Organization aims to encourage code development. We prefer to empower our users to build end-user applications, design maps, run map hosting services or distribute data rather than making it part of the MapLibre Projects.

https://github.com/maplibre/maplibre/blob/main/CHARTER.md

**Comment by wipfli** [comment]
On the technical side of things, I recommend using a cloud-optimized format like PMTiles which works with http range requests if you are looking into storting map tiles on a S3-compatible service.

**Comment by ganganstyle** [comment]
Thank you for the replies @wipfli. Maybe this should instead be a request directed to e.g. AWS or Azure?
It makes more sense to drive profits from services such as route planning, location search etc, than from what should be just static file accessing imo.

PMTiles is a great project, though as I understand it, can't get anywhere near the response times of a pure data response due to script processing requirements?
